test/eqDeep.js

```diff
@@ -0,0 +1,86 @@
+/*jshint typed: true */
+var assert = require('assert');
+
+var R = require('..');
+
+
+describe('eqDeep', function() {
+    var a = [];
+    var b = a;
+    it('tests for deep equality of its operands', function() {
+        assert.strictEqual(R.eqDeep(100, 100), true);
+        assert.strictEqual(R.eqDeep(100, '100'), false);
+        assert.strictEqual(R.eqDeep([], []), true);
+        assert.strictEqual(R.eqDeep(a, b), true);
+    });
+
+    it('handles objects', function() {
+        assert.strictEqual(R.eqDeep({}, {}), true);
+        assert.strictEqual(R.eqDeep({a:1, b:2}, {a:1, b:2}), true);
+        assert.strictEqual(R.eqDeep({a:2, b:3}, {b:3, a:2}), true);
+        assert.strictEqual(R.eqDeep({a:2, b:3}, {a:3, b:3}), false);
+        assert.strictEqual(R.eqDeep({a:2, b:3, c:1}, {a:2, b:3}), false);
+    });
+
+    it('handles regex', function() {
+        assert.strictEqual(R.eqDeep(/\s/, /\s/), true);
+        assert.strictEqual(R.eqDeep(/\s/, /\d/), false);
+        assert.strictEqual(R.eqDeep(/a/gi, /a/ig), true);
+        assert.strictEqual(R.eqDeep(/a/mgi, /a/img), true);
+        assert.strictEqual(R.eqDeep(/a/gi, /a/i), false);
+    });
+
+    var listA = [1, 2, 3];
+    var listB = [1, 3, 2];
+    it('handles lists', function() {
+        assert.strictEqual(R.eqDeep([], {}), false);
+        assert.strictEqual(R.eqDeep(listA, listB), false);
+    });
+
+    var c = {}; c.v = c;
+    var d = {}; d.v = d;
+    var e = []; e.push(e);
+    var f = []; f.push(f);
+    var nestA = {a:[1, 2, {c:1}], b:1};
+    var nestB = {a:[1, 2, {c:1}], b:1};
+    var nestC = {a:[1, 2, {c:2}], b:1};
+    it('handles recursive data structures', function() {
+        assert.strictEqual(R.eqDeep(c, d), true);
+        assert.strictEqual(R.eqDeep(e, f), true);
+        assert.strictEqual(R.eqDeep(nestA, nestB), true);
+        assert.strictEqual(R.eqDeep(nestA, nestC), false);
+    });
+
+    var date1 = new Date(0);
+    var date2 = new Date(0);
+    var date3 = new Date(1);
+    var date4 = new Date(0);
+    date4.v = 'value';
+    it('handles dates', function() {
+        assert.strictEqual(R.eqDeep(date1, date2), true);
+        assert.strictEqual(R.eqDeep(date2, date3), false);
+        assert.strictEqual(R.eqDeep(date1, date4), false);
+    });
+
+    var typArr1 = new ArrayBuffer(10);
```

Comment:
@davidchambers Would there be problems with wrapping that test case in an if?
``` javascript
if (ArrayBuffer) {
    // typed array tests
}
````